src/Microsoft.AspNet.Identity/UserManager.cs

```diff
@@ -1915,10 +1919,26 @@ public virtual Task<IList<TUser>> GetUsersInRoleAsync(string roleName)
         /// <param name="user"></param>
         /// <param name="methodName"></param>
         /// <returns></returns>
-        protected async Task<IdentityResult> LogResultAsync(IdentityResult result,
+        protected virtual async Task<IdentityResult> LogResultAsync(IdentityResult result,
             TUser user, [System.Runtime.CompilerServices.CallerMemberName] string methodName = "")
         {
-            result.Log(Logger, Resources.FormatLoggingResultMessage(methodName, await GetUserIdAsync(user)));
+            if (result.Succeeded)
+            {
+                if (Logger.IsEnabled(LogLevel.Information))
+                {
+                    var baseMessage = Resources.FormatLoggingResultMessage(methodName, await GetUserIdAsync(user));
+                    Logger.WriteInformation(Resources.FormatLogIdentityResultSuccess(baseMessage));
+                }
+            }
+            else
+            {
+                if (Logger.IsEnabled(LogLevel.Warning))
+                {
+                    var baseMessage = Resources.FormatLoggingResultMessage(methodName, await GetUserIdAsync(user));
+                    Logger.WriteWarning(Resources.FormatLogIdentityResultFailure(baseMessage,
+                        string.Join(",", result.Errors.Select(x => x.Code).ToList())));
+                }
+            }
             return result;
         }
 
```

Comment:
The virtual Log method on the IdentityResult was not called by the bool overload anyway so they were present in the last checkin as well. The decision came because we constructed half message in UserManager/RoleManager and rest in the IdentityResult. This make convenient where do the whole construction of the message in one place after checking whether log level is enabled.